driver-scala/src/it/scala/org/mongodb/scala/RequiresMongoDBISpec.scala

```diff
@@ -59,15 +60,22 @@ trait RequiresMongoDBISpec extends BaseSpec with BeforeAndAfterAll {
   }
   val connectionString: ConnectionString = ConnectionString(mongoClientURI)
 
-  def mongoClientSettingsBuilder: MongoClientSettings.Builder =
-    MongoClientSettings.builder().applyConnectionString(connectionString)
+  def mongoClientSettingsBuilder: MongoClientSettings.Builder = {
+    val builder = MongoClientSettings.builder()
+    if (getServerApi != null) {
+      builder.serverApi(getServerApi);
+    }
+    builder
+      .applyConnectionString(connectionString)
+      .codecRegistry(MongoClient.DEFAULT_CODEC_REGISTRY)
```

Comment:
Took me a while to figure out that this was required when constructing a MongoClient with MongoClientSettings instead of a ConnectionString.  Has this bitten users?

Reply:
Are you referring to having to provide the default codec registry?

Reply:
Yes.  For Scala it's required since a codec has to be registered for the Scala driver-specific `Document` classes.  When creating a MongoClient with just a connection string, the driver does that for you.  But with MongoClientSettings it does not.

Reply:
It may have bitten users but they haven't reported it back.  I imagine they either manually add the scala codec registry if they have a custom registry.

Note: there is the `org.mongodb.scala.MongoClientSettings` helper that sets the scala default codec registry when calling the builder method.